### DISCUSSION TITLE
Possible to use a single function from react-native-IAP?

### DISCUSSION THREAD

**Discussion by mj-schmitz** [hyochan/react-native-iap#2128] [Q&A]
Hello, I have implemented expo-IAP in my app and can purchase the subscription.  However, there is no way to check subscription status using EXPO. I am hoping to use this react-native-IAP package so I can simply call subscriptionStatus using storekit2 and check the status.  That is the only thing I wish to implement.

Is it possible to simply call this so I can get a true/false?

Here is the current implementation I have, and I get undefined when trying to read it.:

```
import { useIAP, endConnection,initConnection,IapIosSk2, setup} from "react-native-iap"

async nativeTest() : Promise<any>{
        setup({storekitMode:'STOREKIT2_MODE'})
        await initConnection()
        const status = await (IapIosSk2).subscriptionStatus('sku')

        console.log((JSON.stringify(status) + 'status'))
        await endConnection();

        return status

    }
```

and if I don't stringify, I get the following :

` {"_1":0,"_2":0,"_3":null,"_4":null}status`
or 
`undefinedstatus`

Is there a way to do this without implementing an entire workflow for IAP?

**Comment by andresesfm** [comment]
try getting products/subscriptions before `subscriptionStatus`

**Reply by mj-schmitz** [reply]
I'm able to successfully get both and have implemented the logic for purchasing the subscription via the source code in the subscription screen example. 

Here's what I'm trying now :

`

const isSubbed = () => {

          console.log('awaiting subbed')

          IapIosSk2.subscriptionStatus('subscriotion sku').then((res)=>{

            res.map((result) => {
              console.log('status Success: ',result)
            })
          }).catch( ()=>{
            console.log('failed to get status')
          })
    }
`

and then I render a button to call onpress={isSubbed}

I am now getting returned with failed to get subscription status via the rejected promise.

Am I using the wrong 'sku'? Should it be the subscription sku that is in the app store, or the app sku?

**Comment by mj-schmitz** [comment]
I have fixed what I thought was the problem with my handling of the promise.  Here is the code I would assume to work:

```
const isSubbed = () => {
        console.log('awaiting subbed')
        return IapIosSk2.subscriptionStatus('subscription sku').then(res=>{
          console.log('returned: ', res)
          return res
        }).catch( ()=>{
          console.log('failed to get status')
        })
    }
   
    isSubbed().then(res=>{
      console.log(res + 'res')
    })
```

However, this also gives undefined.  What is the sku I should be putting into the function, this should return the value correct?

**Reply by andresesfm** [reply]
Not sure if you saw my earlier comment but, have you called `getProducts/getSubscriptions` before calling `subscriptionStatus` ?

**Reply by mj-schmitz** [reply]
Yes I saw it, I am able to succesfully get, display, and purchase my subscriptions (essentially the same as the subscription example in the source code) But when I put the above function before the return statement and after the getProducts/getsubscription functions, I get undefined.

**Reply by andresesfm** [reply]
It all comes down to this line: https://github.com/dooboolab/react-native-iap/blob/main/ios/RNIapIosSk2.swift#L762 The reasons I can see `status` being `nil` are that a) `sku` in `products`[sku]` doesn't match the one you used for `getsubscriptions`. b) you didn't call `getSubscriptions hence `products` is `nil` c) The `sku` you are querying is not a subscription. Let us know, I home that this helps

**Reply by mj-schmitz** [reply]
Thank you for the response.  I'm confused however as to why in b) getSubscriptions would fill products? I thought this would fill subscriptions.  I do not have any products but I do have the subscription field and am mapping the results to the screen.  

Is having a product necessary? I am under the assumption that for renewable subscriptions, I only need the getSubscriptions that grabs it from the app store.  I was able to create a test product in app store and get the product details just to make sure, and the status is still coming undefined.  

I can't seem to find any example code of subscriptionStatus, but it seems like what I have should surely work, as I am using the same SKU for getSubscriptions as i am for subscriptionStatus.

**Reply by andresesfm** [reply]
`products` is just the variable name that holds both products and subscriptions on the native module.

Sorry, there's no example of this method as it's new for sk2. It's a good chance for a contribution
